### DISCUSSION TITLE
Auto scroll and focus not working for other input fields than text field

### DISCUSSION THREAD

**Discussion by sbimochan** [react-hook-form/react-hook-form#5925] [Q&A]
**Describe the bug**
On using handleSubmit, if the fields have validation errors, it do work for text fields. but for other fields that needs `Controller`, it neither scrolls not focus

**To Reproduce**

1. Go to 'https://codesandbox.io/s/react-hook-form-handlesubmit-v7-forked-kgm23?file=/src/index.jsx'
2. Click on 'Submit.'
3. Scroll up to 'top'
4. It doesn't scroll to date field but only text field.

**Codesandbox link (Required)**
https://codesandbox.io/s/react-hook-form-handlesubmit-v7-forked-kgm23?file=/src/index.jsx

V7

- [JS](https://codesandbox.io/s/great-paper-deove) Template
- [TS](https://codesandbox.io/s/eager-sun-jt4df) Template

V6

- [JS](https://codesandbox.io/s/mystifying-keller-5jwf5) Template
- [TS](https://codesandbox.io/s/romantic-moon-0chz7) Template

**Expected behavior**
It should at least scroll to all input fields having error

**Comment by bluebill1049** [comment]
this is due to `ref` is not supplied with `Controller`, please refer to the doc for more information.

https://react-hook-form.com/api/usecontroller

<img width="1011" alt="Screen Shot 2021-07-13 at 6 41 23 pm" src="https://user-images.githubusercontent.com/10513364/125420458-d515061a-302c-4f48-ace2-5c5a2aa7eda5.png">

**Reply by sbimochan** [reply]
Thank you for quick reply @bluebill1049 I'll do a quick test.

**Reply by sbimochan** [reply]
@bluebill1049 , There was already `inputRef={ref}` in line 59. That's why it was not working.

**Reply by sbimochan** [reply]
note: The version is 6.15

**Reply by bluebill1049** [reply]
then you need to use onFocus callback refer to v6 doc

**Reply by sbimochan** [reply]
Nope it's not working. I've even tried in onError with ref.current.focus() but it doesn't even focus and there are no other methods like scrollIntoView

**Reply by bluebill1049** [reply]
then the `ref` is definitely not there then, if it works for textField, it means it can invoke the `focus` callback.
